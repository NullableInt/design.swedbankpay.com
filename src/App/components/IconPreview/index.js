import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const IconPreview = ({ name, preview, previewSize, size, className, type }) => {
    const classNames = className ? className.split(" ") : [];
    const iconClasses = classnames(type, type === "payment-icon" ? name : "", size, ...classNames);

    return preview || previewSize ? (
        <div className="icon-preview">
            <i className={iconClasses}>{iconClasses.includes("material-icons") ? name : null}</i>
            <span>{preview ? name : size}</span>
        </div>
    ) : <i className={iconClasses}>{iconClasses.includes("material-icons") ? name : null}</i>;
};

IconPreview.propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    preview: PropTypes.bool,
    size: PropTypes.oneOf([
        "tiny",
        "small",
        "medium",
        "large",
        "huge"
    ]),
    className: PropTypes.string
};

export default IconPreview;
