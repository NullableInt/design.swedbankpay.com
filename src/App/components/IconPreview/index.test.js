import React from "react";
import { shallow } from "enzyme";

import IconPreview from "./index";

describe("Component: IconPreview - ", () => {
    it("is defined", () => {
        expect(IconPreview).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
    });

    it("prop name is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<IconPreview type="payment-icon" />);

        expect(console.error).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it("prop type is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<IconPreview name="amex" />);

        expect(console.error).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it("prop size is an enum with specified allowed values", () => {
        console.error = jest.fn();

        const allowedSizes = shallow(
            <div>
                <IconPreview size="tiny" type="material-icons" name="visa" />
                <IconPreview size="small" type="payment-icon" name="amex" />
                <IconPreview size="medium" type="payment-icon" name="diners" />
                <IconPreview size="large" type="payment-icon" name="mastercard" />
                <IconPreview size="huge" type="payment-icon" name="mobilepay" />
            </div>
        );

        expect(console.error).not.toHaveBeenCalled();

        const illegalSize = shallow(<IconPreview name="amex" size="test" type="payment-icon" />);

        expect(console.error).toHaveBeenCalled();

        expect(allowedSizes).toMatchSnapshot();
        expect(illegalSize).toMatchSnapshot();
    });

    it("renders a material icon when type is equal to material-icons", () => {
        const wrapper = shallow(<IconPreview name="android" type="material-icons" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="material-icons">android</i>)).toEqual(true);
    });

    it("renders correct icon when prop name is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon amex"></i>)).toEqual(true);
    });

    it("renders correct size when prop size is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" size="large" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon amex large"></i>)).toEqual(true);
    });

    it("renders icon with custom when prop className is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" className="test test2 test-3" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon amex test test2 test-3"></i>)).toEqual(true);
    });

    it("renders a preview wrapper around payment-icon when prop preview is true", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" preview />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(
            <div className="icon-preview">
                <i className="payment-icon amex"></i>
                <span>amex</span>
            </div>
        )).toEqual(true);
    });

    it("renders a preview wrapper around material-icon when prop preview is true", () => {
        const wrapper = shallow(<IconPreview name="android" type="material-icons" preview />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(
            <div className="icon-preview">
                <i className="material-icons">android</i>
                <span>android</span>
            </div>
        )).toEqual(true);
    });

    it("renders a preview wrapper around material-icon when prop previewSize is true with a span containing the size as text", () => {
        const wrapper = shallow(<IconPreview name="android" type="material-icons" size="large" previewSize />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(
            <div className="icon-preview">
                <i className="material-icons large">android</i>
                <span>large</span>
            </div>
        )).toEqual(true);
    });
});
