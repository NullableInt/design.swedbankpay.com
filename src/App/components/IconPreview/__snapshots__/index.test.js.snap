// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Panel -  prop name is an enum with specified allowed values 1`] = `
<div>
  <PaymentIcon
    name="amex"
  />
  <PaymentIcon
    name="diners"
  />
  <PaymentIcon
    name="mastercard"
  />
  <PaymentIcon
    name="mobilepay"
  />
  <PaymentIcon
    name="paypal"
  />
  <PaymentIcon
    name="swish"
  />
  <PaymentIcon
    name="vipps"
  />
  <PaymentIcon
    name="visa"
  />
</div>
`;

exports[`Component: Panel -  prop name is an enum with specified allowed values 2`] = `
<i
  className="payment-icon test"
/>
`;

exports[`Component: Panel -  prop name is marked as required 1`] = `
<i
  className="payment-icon"
/>
`;

exports[`Component: Panel -  prop size is an enum with specified allowed values 1`] = `
<div>
  <PaymentIcon
    name="amex"
    size="small"
  />
  <PaymentIcon
    name="diners"
    size="medium"
  />
  <PaymentIcon
    name="mastercard"
    size="large"
  />
  <PaymentIcon
    name="mobilepay"
    size="huge"
  />
</div>
`;

exports[`Component: Panel -  prop size is an enum with specified allowed values 2`] = `
<i
  className="payment-icon amex test"
/>
`;

exports[`Component: Panel -  renders 1`] = `
<i
  className="payment-icon amex"
/>
`;

exports[`Component: Panel -  renders a preview wrapper around icon when prop preview is true 1`] = `
<div
  className="icon-preview"
>
  <i
    className="payment-icon amex"
  />
  <span>
    amex
  </span>
</div>
`;

exports[`Component: Panel -  renders correct icon when prop name is provided 1`] = `
<i
  className="payment-icon amex"
/>
`;

exports[`Component: Panel -  renders correct size when prop size is provided 1`] = `
<i
  className="payment-icon amex large"
/>
`;

exports[`Component: Panel -  renders icon with custom when prop className is provided 1`] = `
<i
  className="payment-icon amex test test2 test-3"
/>
`;
